### PR TITLE
Actually run Hacking rules as part of PEP8 target

### PR DESCRIPTION
**Pull request by JordanP**
For some reason, Hacking was not installed by the pep8 tox target.
This patch makes sure Hacking is installed and run when pep8 tests
are run. Also this patch fixes a couple of violations.

### CODE REVIEW

File: test/unit/test_ranges.py [hunk 1]
```diff
@@ -46,10 +46,8 @@
 
 
 class TestRangeHeaders(unittest.TestCase):
-    """
-    Test outgoing range headers to sproxyd from the object server on a partial
-    get of an object. See
-    https://github.com/scality/ScalitySproxydSwift/issues/121 for more details.
+    """Test outgoing range headers to sproxyd from the object server on a partial
+    get of an object. See ScalitySproxydSwift issue #121 for more details.
```
Comment (NicolasT) [inline test/unit/test_ranges.py]: I'd keep the URL to the issue somewhere in here. Much easier to browse to and such. Maybe as a reference in a second paragraph, or use ReST-like formatting?

File: test/unit/test_server.py [hunk 2]
```diff
@@ -34,8 +34,7 @@
 
 
 def test_api_compatible():
-    '''
-    Test whether `swift_scality_backend.server.app_factory`'s result is
+    '''Test whether `swift_scality_backend.server.app_factory`'s result is
```
Comment (NicolasT) [inline test/unit/test_server.py]: Hmpf, not very fond of this 'rule' then... If something fits on a single line, having it at the same line as the quotes, OK. But this looks a bit distracting.

File: tox.ini [hunk 3]
```diff
@@ -27,7 +27,7 @@ whitelist_externals =
   echo
 install_command = echo {packages}
 commands =
-  pip install flake8
+  pip install -r test-requirements.txt
```
Comment (NicolasT) [inline tox.ini]: Couldn't this be the `install_command`?

File: test/unit/test_server.py (hunk 2 above)
Comment (JordanP) [inline test/unit/test_server.py]: I don't like this rule either: dura lex, sed lex

In most case it makes sense though, so I don"t want to make an exception and I can live with this weird doc string here.

File: tox.ini (hunk 3 above)
Comment (JordanP) [inline tox.ini]: I tried but I got "tox.ConfigError: ConfigError: 'install_command' must contain '{packages}' substitution"

Comment (NicolasT) [inline tox.ini]: Didn't try, but what about
```
install_command =
    echo Not installing {package}
    pip install -r test-requirements.txt
```

**Comment by JordanP** [comment]
This works but I got a lot of new errors like 'D300  Use """triple double quotes""" (found '''-quotes)
' or 'D105  Missing docstring in magic method' I have never seen this error in any openstack related projects and I don"t now where they come from !

**Pull request closed without merging**